### ISSUE TITLE
Rayleigh-Bernard 3D example fails if the pressure analysis task is removed.

### ISSUE THREAD

**Issue by csskene** [DedalusProject/dedalus#115]
I've found that running the 3D Rayleigh-Bernard example `examples/ivp/3d_rayleigh_benard/rayleigh_benard.py` in parallel with line 109 commented out
```
# snap.add_task("interp(p, z=0)", scales=1, name='p midplane')
```
 i.e. the analysis task for pressure is removed, then the code fails with the following error 
```
Traceback (most recent call last):
  File "rayleigh_benard.py", line 133, in <module>
    solver.step(dt)
  File "/Users/calumskene/opt/miniconda3/envs/dedalus-test/lib/python3.8/site-packages/dedalus/core/solvers.py", line 518, in step
    path.increment(self.state.fields)
  File "/Users/calumskene/opt/miniconda3/envs/dedalus-test/lib/python3.8/site-packages/dedalus/core/distributor.py", line 441, in increment
    self.increment_group(*fields)
  File "/Users/calumskene/opt/miniconda3/envs/dedalus-test/lib/python3.8/site-packages/dedalus/core/distributor.py", line 490, in increment_group
    scales = unify(field.scales for field in fields)
  File "/Users/calumskene/opt/miniconda3/envs/dedalus-test/lib/python3.8/site-packages/dedalus/tools/general.py", line 74, in unify
    raise ValueError("Objects are not all equal.")
ValueError: Objects are not all equal.
```
Although it seems to work in series.

**Comment by kburns** [comment]
I think this problem was fixed in f90efeee18f5f5f667233db7467e46932e66d9e3.  If you install from the master git branch, it should now work, and the patch will be in the next pip release.

**Comment by csskene** [comment]
I can confirm that installing Dedalus from the master git branch fixes this issue. Thanks very much.
